app/services/auto_end_previous_reservation.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class AutoEndPreviousReservation
+
+  def initialize(product, current_user)
+    @product = product
+    @current_user = current_user
+  end
+
+  def end_previous_reservations!
+    return unless SettingsHelper.feature_on?(:auto_end_reservations_on_next_start)
```

Comment:
this is also checked on the instrument switcher, does it make sense to have both?